src/mcharsetdetector.cpp

```diff
@@ -500,6 +502,12 @@ QList<MCharsetMatch> MCharsetDetector::detectAll()
 QString MCharsetDetector::text(const MCharsetMatch &charsetMatch)
 {
     Q_D(MCharsetDetector);
+    #if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+    if(charsetMatch.name().toLatin1().isEmpty() || charsetMatch.name().toLatin1() != "UTF-8") {
+        qWarning() << "Don`t use non utf8 charset";
```

Comment:
Wouldn't it make the class detecting encodings a bit pointless if it can be used only with utf8? Could use core5compat and use the QTextCodec?